sdks/java/core/src/main/java/org/apache/beam/sdk/metrics/MetricsContainer.java

```diff
@@ -0,0 +1,150 @@
+      new MetricsMap<>(new MetricsMap.Factory<MetricName, CounterCell>() {
+        @Override
+        public CounterCell createInstance(MetricName unusedKey) {
+          return new CounterCell();
+        }
+      });
+
+  private MetricsMap<MetricName, DistributionCell> distributions =
+      new MetricsMap<>(new MetricsMap.Factory<MetricName, DistributionCell>() {
+        @Override
+        public DistributionCell createInstance(MetricName unusedKey) {
+          return new DistributionCell();
+        }
+      });
+
+  /**
+   * Create a new {@link MetricsContainer} associated with the given {@code stepName}.
+   */
+  public MetricsContainer(String stepName) {
+    this.stepName = stepName;
+  }
+
+  /**
+   * Return the {@link CounterCell} that should be used for implementing the given
+   * {@code metricName} in this container.
+   */
+  public CounterCell getCounter(MetricName metricName) {
+    return counters.get(metricName);
+  }
+
+  public DistributionCell getDistribution(MetricName metricName) {
+    return distributions.get(metricName);
+  }
+
+  private <UpdateT, CellT extends MetricCell<?, UpdateT>>
+  ImmutableList<MetricUpdate<UpdateT>> extractUpdates(
+      MetricsMap<MetricName, CellT> cells) {
+    ImmutableList.Builder<MetricUpdate<UpdateT>> updates = ImmutableList.builder();
+    for (Map.Entry<MetricName, CellT> cell : cells.entries()) {
+      if (cell.getValue().getDirty().beforeCommit()) {
+        updates.add(MetricUpdate.create(MetricKey.create(stepName, cell.getKey()),
+            cell.getValue().getCumulative()));
+      }
+    }
+    return updates.build();
+  }
+
+  /**
+   * Return the cumulative values for any metrics that have changed since the last time updates were
+   * committed.
+   */
+  public MetricUpdates getUpdates() {
+    return MetricUpdates.create(
+        extractUpdates(counters),
+        extractUpdates(distributions));
+  }
+
+  private void commitUpdates(MetricsMap<MetricName, ? extends MetricCell<?, ?>> cells) {
+    for (MetricCell<?, ?> cell : cells.values()) {
+      cell.getDirty().afterCommit();
+    }
+  }
+
+  /**
+   * Mark all of the updates that were retrieved with the latest call to {@link #getUpdates()} as
+   * committed.
+   */
+  public void commitUpdates() {
+    commitUpdates(counters);
+    commitUpdates(distributions);
+  }
+
+  private <UpdateT, CellT extends MetricCell<?, UpdateT>>
+  ImmutableList<MetricUpdate<UpdateT>> extractCumulatives(
+      MetricsMap<MetricName, CellT> cells) {
+    ImmutableList.Builder<MetricUpdate<UpdateT>> updates = ImmutableList.builder();
+    for (Map.Entry<MetricName, CellT> cell : cells.entries()) {
+      UpdateT update = checkNotNull(cell.getValue().getCumulative());
+      updates.add(MetricUpdate.create(MetricKey.create(stepName, cell.getKey()), update));
+    }
+    return updates.build();
+  }
+
+  /**
+   * Return the {@link MetricUpdates} representing the cumulative values of all metrics in this
+   * container.
+   */
```

Comment:
Hmm... It looks like this is dead code. It was refactored into the `extractCumulatives(counters)` and `extractCumulatives(distributions)`.